File: ChangeTheme_OC/ChangeTheme_OC/App/UIStyle/ArtUIStyleManager+UIStyleApp.m
//
//  ArtUIStyleManager+UIStyleApp.m
//  ChangeTheme_OC
//
//  Created by weijingyun on 2017/8/24.
//  Copyright © 2017年 weijingyun. All rights reserved.
//

#import "ArtUIStyleManager+UIStyleApp.h"

@implementation ArtUIStyleManager (UIStyleApp)

- (NSString *)getStyleName_App {
    return @"AppStyle.plist";
}

@end


@implementation UIColor (UIStyleApp)

+ (UIColor *)artAppForKey:(NSString *)aColorKey {
    return [self artModule:@"App_Main" colorForKey:aColorKey];
}

+ (void)artAppForKey:(NSString *)aColorKey strongSelf:(id)strongSelf block:(void(^)(UIColor *color, id weakSelf))aBlock {
    [self artModule:@"App_Main" colorForKey:aColorKey strongSelf:strongSelf block:aBlock];
}

@end

@implementation UIFont (UIStyleCourseware)

+ (UIFont *)artAppForKey:(NSString *)aFontKey {
    return [self artModule:@"App_Main" fontForKey:aFontKey];
}

+ (void)artAppForKey:(NSString *)aFontKey strongSelf:(id)strongSelf block:(void(^)(UIFont *font, id weakSelf))aBlock {
    [self artModule:@"App_Main" fontForKey:aFontKey strongSelf:strongSelf block:aBlock];
}

@end

File: ChangeTheme_Swift/ChangeTheme_Swift/App/UIStyle/ArtUIStyleManager+UIStyleApp.swift
//
//  ArtUIStyleManager+UIStyleApp.swift
//  ChangeTheme_Swift
//
//  Created by weijingyun on 2017/9/5.
//  Copyright © 2017年 weijingyun. All rights reserved.
//

import UIKit

extension ArtUIStyleManager {
    
    func getStyleName_App() -> String! {
        return "AppStyle.plist"
    }
    
    func getHotReloaderStylePath_App() -> String! {
        return "/App/UIStyle/AppStyle.plist";
    }
    
}

extension ArtUIStyle {
    class func artAppStyle(styleKey : String) -> ArtUIStyle {
        return ArtUIStyleManager.artStyle(module: "App_Main", styleKey:styleKey)
    }
}

extension UIColor {
    class func artAppStyle(styleKey : String) -> UIColor {
        return ArtUIStyleManager.artStyle(module: "App_Main", styleKey:styleKey).color()
    }
}

extension UIFont {
    class func artAppStyle(styleKey : String) -> UIFont {
        return ArtUIStyleManager.artStyle(module: "App_Main", styleKey:styleKey).font()
    }
}

extension UIImage {
    class func artAppImage(imageString : String) -> UIImage {
        return ArtUIStyleManager.artImage(module: "App_Main", imageString: imageString).image()
    }
}
